### ISSUE TITLE
REDMINE_RELATIVE_URL_ROOT breaks nginx handling files 

### ISSUE THREAD

**Issue by jcormier** [sameersbn/docker-redmine#240]
The extra relative root part gets prepended to the file path so nginx tries to find file at `/home/redmine/redmine/public/redmine/stylesheets/jquery/jquery-ui-1.11.0.css.html` instead of `/home/redmine/redmine/public/stylesheets/jquery/jquery-ui-1.11.0.css.html`

Note that nothing noticeably breaks but the file ends up being served by unicorn instead of nginx.  Kinda defeats the purpose.

Excerpt from debug nginx log
```
  576 2016/11/30 13:46:43 [debug] 327#327: *1 try files phase: 13
  577 2016/11/30 13:46:43 [debug] 327#327: *1 http script var: "/redmine/stylesheets/jquery/jquery-ui-1.      11.0.css"
  578 2016/11/30 13:46:43 [debug] 327#327: *1 trying to use file: "/redmine/stylesheets/jquery/jquery-ui      -1.11.0.css" "/home/redmine/redmine/public/redmine/stylesheets/jquery/jquery-ui-1.11.0.css"       
  579 2016/11/30 13:46:43 [debug] 327#327: *1 trying to use file: "index.html" "/home/redmine/redmine/pu      blicindex.html"
  580 2016/11/30 13:46:43 [debug] 327#327: *1 http script var: "/redmine/stylesheets/jquery/jquery-ui-1.      11.0.css"
  581 2016/11/30 13:46:43 [debug] 327#327: *1 http script copy: ".html"
  582 2016/11/30 13:46:43 [debug] 327#327: *1 trying to use file: "/redmine/stylesheets/jquery/jquery-ui      -1.11.0.css.html" "/home/redmine/redmine/public/redmine/stylesheets/jquery/jquery-ui-1.11.0.css.ht      ml"
  583 2016/11/30 13:46:43 [debug] 327#327: *1 trying to use file: "@redmine" "/home/redmine/redmine/publ      ic@redmine"
  584 2016/11/30 13:46:43 [debug] 327#327: *1 test location: "@redmine"
  585 2016/11/30 13:46:43 [debug] 327#327: *1 using location: @redmine "/redmine/stylesheets/jquery/jque      ry-ui-1.11.0.css?"

```

**Comment by jcormier** [comment]
Here's my first attempt to address the issue and add the cache expires header.  It works but not sure if it can be done easier. 

Modified from this link
http://www.redmine.org/projects/redmine/wiki/BrowserCaching

```
+   # the regex logic:   after either /javascripts/ or /stylesheets/ find the suffixes we want, followed by any quantity of numbers 0-9
+   # This works because the files we want to cache always appear after one of those 2 directories: but not the files we want to ignore
+   # /journals/edit/24174.js   and /uploads.js?attachment_id=1&filename=my-file-to-upload.png
+   location ~* /redmine(?<file>/(?:(?:plugin_assets/|themes/).+)?(?:javascripts|stylesheets|images)/.+(?:css|js|jpe?g|gif|ico|png)(\?[0-9]+)?$) {
+ 
+     expires 365d;
+     gzip on;
+     gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;gzip_types text/plain text/css application/json application/javascript text/xml application/xml
+ 
+     try_files $file @redmine;
+   }
```

**Comment by jcormier** [comment]
Here's my first attempt to address the issue and add the cache expires header.  It works but not sure if it can be done easier. 

Modified from this link
http://www.redmine.org/projects/redmine/wiki/BrowserCaching

```
+   # the regex logic:   after either /javascripts/ or /stylesheets/ find the suffixes we want, followed by any quantity of numbers 0-9
+   # This works because the files we want to cache always appear after one of those 2 directories: but not the files we want to ignore
+   # /journals/edit/24174.js   and /uploads.js?attachment_id=1&filename=my-file-to-upload.png
+   location ~* /redmine(?<file>/(?:(?:plugin_assets/|themes/).+)?(?:javascripts|stylesheets|images)/.+(?:css|js|jpe?g|gif|ico|png)(\?[0-9]+)?$) {
+ 
+     expires 365d;
+     gzip on;
+     gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;gzip_types text/plain text/css application/json application/javascript text/xml application/xml
+ 
+     try_files $file @redmine;
+   }
```

**Comment by jcormier** [comment]
@sameersbn @frank-dspeed 
Thoughts on this change?

**Comment by frank-dspeed** [comment]
@jcormier in general i think nginx should always serve the static content so this is a good change that addresses also performance issues 

LGTM +1

**Comment by frank-dspeed** [comment]
@jcormier we should solve that on a other way via a extra location probally that does solve that we can try on multiple locations also i will think about that more deep later but for the first time to get it done

LGTM +1 

and rename this issue to something like all files should be served via nginx directly

**Comment by frank-dspeed** [comment]
@sameersbn @jcormier we need to apply labels to this project so we can sort issues more well.

**Comment by jcormier** [comment]
Hmm @sameersbn marked a issue stale so I assume there are labels but that I simply don''t have permission to set them.

**Comment by frank-dspeed** [comment]
@jcormier yes thats why i write it here so when he comes here he will read that up and fix that as its importent for our issue handling.

**Comment by jcormier** [comment]
Just noticed that my gzip_types line overrides itself.

    +     gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;gzip_types text/plain text/css application/json application/javascript text/xml application/xml;

Actually has two gzip_types in there.

    gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;
    gzip_types text/plain text/css application/json application/javascript text/xml application/xml;

Its been working for me like this but I'll probably remove the extra line.

**Comment by sameersbn** [comment]
@jcormier I am okay with these changes. Please create a pull-request for the change and lets proceed.

**Comment by frank-dspeed** [comment]
@jcormier i reviewed this change now and i have many questions about it this is adding cache and should fix issue #240 wich is about something totally diffrent?

**Comment by jcormier** [comment]
Hmm good point. My commit does attempt to change/fix two things at once and only one of them is relevant to this issue.

Will look into reworking my cache/gzip change as a seperate commit.
